### PR TITLE
[Fix #218] Add event dispatch management

### LINKED ISSUE #218: Move event dispatcher management inside
Up to this point, if a user wants to listen to events from sumo activities, it has to provide its own `gen_event` dispatcher.
It would be a much better idea to just add a `sumo_events_sup` supervisor with a `one_for_one` strategy that basically boots up a `gen_event` dispatcher per each dispatcher listed on the `events` section of the app configuration.

### PR DESCRIPTION
**Pull request by harenson**

### CODE REVIEW

File: test/sumo_events_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+-spec end_per_suite(config()) -> config().
+end_per_suite(Config) ->
+  Config.
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Common test
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec events_manager_supervisor_running(Config::config()) -> {comment, string()}.
```
Comment (gadgetci) [inline test/sumo_events_SUITE.erl]: According to **Elvis**:
> Line 49 is too long: -spec events_manager_supervisor_running(Config::config()) -> {comment, string()}..

Comment (gadgetci) [inline test/sumo_events_SUITE.erl]: According to **Elvis**:
> Line 49 is too long: -spec events_manager_supervisor_running(Config::config()) -> {comment, string()}..

File: src/sumo_event_manager_sup.erl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+-behaviour(supervisor).
+
+-export([start_link/0]).
+-export([init/1]).
+
+-spec start_link() -> {ok, pid()} | {error, term()}.
+start_link() -> supervisor:start_link({local, ?MODULE}, ?MODULE, []).
+
+-spec init(any()) ->
```
Comment (gadgetci) [inline src/sumo_event_manager_sup.erl]: According to **Dialyzer**:
>  The return type {'ok',{_,[{_,{atom() | tuple(),atom(),'undefined' | [any()]},'permanent' | 'temporary' | 'transient','brutal_kill' | 'infinity' | non_neg_integer(),'supervisor' | 'worker','dynamic' | [atom() | tuple()]}]}} in the specification of init/1 is not a subtype of 'ignore' | {'ok',{{'one_for_all',non_neg_integer(),non_neg_integer()} | {'one_for_one',non_neg_integer(),non_neg_integer()} | {'rest_for_one',non_neg_integer(),non_neg_integer()} | {'simple_one_for_one',non_neg_integer(),non_neg_integer()},[{_,{atom() | tuple(),atom(),'undefined' | [any()]},'permanent' | 'temporary' | 'transient','brutal_kill' | 'infinity' | non_neg_integer(),'supervisor' | 'worker','dynamic' | [atom() | tuple()]}]}}, which is the expected return type for the callback of supervisor behaviour

File: test/sumo_events_SUITE.erl [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      false -> ok;
+      {Module, EventManager} ->
+        ct:comment("~p should be running", [EventManager]),
+        [] = gen_event:which_handlers(EventManager)
+    end
+  end,
+  lists:foreach(Fun, sumo_test_utils:all_people()),
+
+  {comment, ""}.
+
```
Comment (elbrujohalcon) [inline test/sumo_events_SUITE.erl]: too many white lines

File: test/sumo_events_SUITE.erl [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                       find_all_sort
+                      ],
+  [F || {F, _} <- Exports, not lists:member(F, ExcludedFunctions)].
+
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) ->
+  sumo_test_utils:start_apps(),
+  Config.
+
+init_per_testcase(_, Config) ->
```
Comment (elbrujohalcon) [inline test/sumo_events_SUITE.erl]: remove these functions, they're not needed

File: test/test.config [hunk 5]
```diff
@@ -101,7 +101,12 @@
      ]
     },
     {events,
-     []
+      [{sumo_test_people_mysql, sumo_test_people_mysql_events_manager},
+       {sumo_test_people_mongo, sumo_test_people_mongo_events_manager},
+       {sumo_test_people_elasticsearch, sumo_test_people_elasticsearch},
```
Comment (elbrujohalcon) [inline test/test.config]: elasticsearch_manager

**Pull request merged**